File: php/src/Advertisement/Application/Command/RenewAdvertisement/RenewAdvertisementUseCase.php
<?php
declare(strict_types=1);

namespace Demo\App\Advertisement\Application\Command\RenewAdvertisement;

use Demo\App\Advertisement\Domain\AdvertisementRepository;
use Demo\App\Advertisement\Domain\Model\ValueObject\Password;
use Exception;

final class RenewAdvertisementUseCase
{
    public function __construct(private AdvertisementRepository $advertisementRepository)
    {
    }

    /**
     * @throws Exception
     */
    public function execute(RenewAdvertisementCommand $command): void
    {
        $advertisement = $this->advertisementRepository->findById($command->id);

        if (!$advertisement->password()->isValidatedWith($command->password)) {
            return;
        }

        $advertisement->renew(Password::fromPlainPassword($command->password));

        $this->advertisementRepository->save($advertisement);
    }
}

File: kotlin/src/main/kotlin/advertisement/application/renewAdvertisement/RenewAdvertisementUseCase.kt
package advertisement.application.renewAdvertisement

import advertisement.domain.AdvertisementRepository
import advertisement.domain.model.value_object.Password
import advertisement.infrastructure.PasswordHasher

class RenewAdvertisementUseCase(private val advertisementRepository: AdvertisementRepository, private val passwordHasher: PasswordHasher) {
    fun execute(renewAdvertisementCommand: RenewAdvertisementCommand) {
        val advertisement = advertisementRepository.findById(renewAdvertisementCommand.id)

        if (!advertisement.password?.validateWith(renewAdvertisementCommand.password, passwordHasher)!!)
            return

        advertisement.renew(Password.fromPlainPassword(renewAdvertisementCommand.password, passwordHasher))

        advertisementRepository.save(advertisement)
    }
}
